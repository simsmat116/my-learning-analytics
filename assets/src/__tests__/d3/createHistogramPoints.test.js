import createHistogram from '../../components/d3/createHistogram'
import { histogramData } from '../testData/d3TestDataPoints'

describe('createHistogram', () => {
  test('should build a histogram bar chart using points data and correctly show points and not percentages', () => {
    const div = document.createElement('div')
    createHistogram({
      data: histogramData,
      gradingType: 'PT',
      width: 1000,
      height: 500,
      domElement: div,
      myGrade: 195.4,
      maxGrade: 279.3,
      showNumberOnBars: false
    })
    expect(div).toMatchSnapshot()
  })

})
