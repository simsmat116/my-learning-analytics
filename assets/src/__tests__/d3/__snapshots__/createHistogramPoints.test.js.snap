// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createHistogram should build a histogram bar chart using points data and correctly show points and not percentages 1`] = `
<div>
  <svg
    height="430"
    width="940"
  >
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285715"
        x="41"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285715"
        x="56.714285714285715"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285708"
        x="72.42857142857143"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285722"
        x="88.14285714285714"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285708"
        x="103.85714285714286"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285708"
        x="119.57142857142857"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285722"
        x="135.28571428571428"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285722"
        x="151"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285722"
        x="166.71428571428572"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285694"
        x="182.42857142857144"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285722"
        x="198.14285714285714"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285694"
        x="213.85714285714286"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.71428571428575"
        x="229.57142857142856"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285694"
        x="245.2857142857143"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285666"
        x="261"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.71428571428578"
        x="276.71428571428567"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285666"
        x="292.42857142857144"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.71428571428578"
        x="308.1428571428571"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285722"
        x="323.8571428571429"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285666"
        x="339.5714285714286"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285722"
        x="355.2857142857143"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285722"
        x="371"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="45"
        width="14.714285714285666"
        x="386.7142857142857"
        y="335"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285722"
        x="402.4285714285714"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.71428571428578"
        x="418.1428571428571"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="90"
        width="14.714285714285722"
        x="433.8571428571429"
        y="290"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285666"
        x="449.5714285714286"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="135"
        width="14.714285714285722"
        x="465.2857142857143"
        y="245"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.71428571428578"
        x="481"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="90"
        width="14.714285714285609"
        x="496.7142857142858"
        y="290"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="45"
        width="14.714285714285722"
        x="512.4285714285713"
        y="335"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="45"
        width="14.71428571428578"
        x="528.1428571428571"
        y="335"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="135"
        width="14.714285714285666"
        x="543.8571428571429"
        y="245"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285666"
        x="559.5714285714286"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="90"
        width="14.71428571428578"
        x="575.2857142857142"
        y="290"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="45"
        width="14.71428571428578"
        x="591"
        y="335"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="270"
        width="14.714285714285666"
        x="606.7142857142858"
        y="110"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="45"
        width="14.71428571428578"
        x="622.4285714285714"
        y="335"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285552"
        x="638.1428571428572"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="135"
        width="14.71428571428578"
        x="653.8571428571428"
        y="245"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285666"
        x="669.5714285714286"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="135"
        width="14.71428571428578"
        x="685.2857142857142"
        y="245"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="45"
        width="14.71428571428578"
        x="701"
        y="335"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="270"
        width="14.714285714285666"
        x="716.7142857142858"
        y="110"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="315"
        width="14.71428571428578"
        x="732.4285714285714"
        y="65"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285552"
        x="748.1428571428572"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="270"
        width="14.71428571428578"
        x="763.8571428571428"
        y="110"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285666"
        x="779.5714285714286"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="135"
        width="14.71428571428578"
        x="795.2857142857142"
        y="245"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.71428571428578"
        x="811"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="14.714285714285666"
        x="826.7142857142858"
        y="380"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="45"
        width="14.71428571428578"
        x="842.4285714285714"
        y="335"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="90"
        width="14.714285714285552"
        x="858.1428571428572"
        y="290"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="360"
        width="14.71428571428578"
        x="873.8571428571428"
        y="20"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="45"
        width="14.714285714285666"
        x="889.5714285714286"
        y="335"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="180"
        width="14.71428571428578"
        x="905.2857142857142"
        y="200"
      />
    </g>
    <g>
      <rect
        fill="steelblue"
        height="0"
        width="0"
        x="921"
        y="380"
      />
    </g>
    <g
      fill="none"
      font-family="sans-serif"
      font-size="10"
      text-anchor="middle"
      transform="translate(0, 380)"
    >
      <path
        class="domain"
        d="M40.5,0.5H920.5"
        stroke="currentColor"
      />
      <g
        class="tick"
        opacity="1"
        transform="translate(449.0714285714286,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          130%
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(480.5,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          140%
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(511.9285714285714,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          150%
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(543.3571428571429,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          160%
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(574.7857142857142,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          170%
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(606.2142857142858,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          180%
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(637.6428571428572,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          190%
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(669.0714285714286,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          200%
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(700.5,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          210%
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(731.9285714285714,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          220%
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(763.3571428571428,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          230%
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(794.7857142857142,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          240%
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(826.2142857142858,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          250%
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(857.6428571428572,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          260%
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(889.0714285714286,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          270%
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(920.5,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          280%
        </text>
      </g>
      <text
        dy="-4"
        fill="rgba(0, 0, 0, 0.87)"
        font-size="0.875rem"
        font-weight="400"
        line-height="1.46429em"
        x="470"
        y="40"
      />
    </g>
    <g
      class="axis"
      fill="none"
      font-family="sans-serif"
      font-size="10"
      text-anchor="end"
      transform="translate(40,0)"
    >
      <g
        class="tick"
        opacity="1"
        transform="translate(0,380.5)"
      >
        <line
          stroke="currentColor"
          x2="940"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-3"
        >
          0
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,290.5)"
      >
        <line
          stroke="currentColor"
          x2="940"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-3"
        >
          2
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,200.5)"
      >
        <line
          stroke="currentColor"
          x2="940"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-3"
        >
          4
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,110.5)"
      >
        <line
          stroke="currentColor"
          x2="940"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-3"
        >
          6
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,20.5)"
      >
        <line
          stroke="currentColor"
          x2="940"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-3"
        >
          8
        </text>
        <text
          dy="-4"
          fill="rgba(0, 0, 0, 0.87)"
          font-size="0.875rem"
          font-weight="400"
          line-height="1.46429em"
          opacity="1"
          text-anchor="start"
          x="4"
        />
      </g>
    </g>
    <line
      stroke="darkorange"
      stroke-width="2"
      transform="translate(0, 380)"
      x1="654.1142857142858"
      x2="654.1142857142858"
      y1="-355"
      y2="0"
    />
    <text
      font-size="0.875rem"
      font-weight="bold"
      line-height="1.46429em"
      text-anchor="start"
      x="534.1142857142858"
      y="35"
    >
      My Grade: 195.4%
    </text>
    <line
      stroke="rgba(0, 0, 0, 0.54)"
      stroke-dasharray="3, 3"
      transform="translate(0, 380)"
      x1="401.4285714285714"
      x2="401.4285714285714"
      y1="-430"
      y2="0"
    />
    <text
      fill="rgba(0, 0, 0, 0.54)"
      font-size="0.875rem"
      line-height="1.46429em"
      text-anchor="start"
      x="361.4285714285714"
      y="15"
    >
      &lt;125%
    </text>
  </svg>
</div>
`;
